### PR TITLE
BUG-7530 - Portal landing page_DOB & Created date alignment issue

### PR DESCRIPTION
**Pull request by nickwhm**
Have changed the layout of the claims list based on new designs. This keeps the details in a summary list https://design-system.service.gov.uk/components/summary-list/. 

The main code changes are lines 164 -204 and lines 124 -161 on the src\components\templates\ClaimsList\index.tsx.

### CODE REVIEW

**Review by gautamkumar-hmrc — APPROVED** [review]

**Comment by IanOvenden** [comment]
We should look to closer adherence to the [GDS summary list](https://design-system.service.gov.uk/components/summary-list/) or use a different format.  Can we review the design with UX and ensure the correct format before moving foward.

It could be there are technical constraints.

**Comment by nickwhm** [comment]
Have changed the layout of the claims list based on new designs. This keeps the details in a summary list https://design-system.service.gov.uk/components/summary-list/.

File: src/components/templates/ClaimsList/index.tsx [hunk 1]
```diff
@@ -25,132 +25,190 @@ export default function ClaimsList(props){
...
+                {child.firstName} {child.lastName}
+              </dd>
+              <dd className='govuk-summary-list__actions govuk-!-width-one-third govuk-!-padding-bottom-0'>
+                {/* If this is the first entry add the status */}
+                {index === 0 ? (
+                  <strong className={`govuk-tag govuk-tag--${claimItem.status.tagColour}`}>
+                    {claimItem.status.text}
+                  </strong>
+                ) : (
+                  index !== 0 && <span className='govuk-visually-hidden'>No action</span>
```
[the remark this answers is not in the document]
Comment (nickwhm) [reply src/components/templates/ClaimsList/index.tsx]: I thought I'd removed that already. It's gone now.

File: src/components/templates/ClaimsList/index.tsx [hunk 2]
```diff
@@ -25,132 +25,190 @@ export default function ClaimsList(props){
...
-                  {child.dob && <><br/><span className='govuk-!-font-weight-regular'>{`${t('DATE_OF_BIRTH')} ${child.dob}`}</span><br/></>}
-                  <span className='govuk-!-font-weight-regular'>{t('CREATED_DATE')} {claimItem.dateCreated}</span>
-                </p>
-              )}
-              {claimItem.actionButton}
+  function renderChildDetails(claimItem) {
+    return claimItem.children.map((child, index) => (
+      <dl className='govuk-summary-list' key={Math.random()}>
+        <div className='govuk-summary-list__row govuk-summary-list__row--no-border'>
+          {child.firstName && (
```
Comment (gautamkumar-hmrc) [inline src/components/templates/ClaimsList/index.tsx:128]: Here you are checking child.firstName only but below line no 134, you are using child.lastName without optional operator.

Comment (nickwhm) [reply src/components/templates/ClaimsList/index.tsx]: Good spot. That check is now unnecessary, so I've removed it.

File: src/components/templates/ClaimsList/index.tsx [hunk 3]
```diff
@@ -25,132 +25,186 @@ export default function ClaimsList(props){
...
-                <p key={child.firstName}>
-                  {child.firstName && child.lastName && `${child.firstName} ${child.lastName}`}
-                  {child.dob && <><br/><span className='govuk-!-font-weight-regular'>{`${t('DATE_OF_BIRTH')} ${child.dob}`}</span><br/></>}
-                  <span className='govuk-!-font-weight-regular'>{t('CREATED_DATE')} {claimItem.dateCreated}</span>
-                </p>
-              )}
-              {claimItem.actionButton}
+  function renderChildDetails(claimItem) {
+    return claimItem.children.map((child, index) => (
+      <dl className='govuk-summary-list' key={Math.random()}>
```
Comment (gautamkumar-hmrc) [inline src/components/templates/ClaimsList/index.tsx:126]: You should use index with child name instead of Math.random().

Comment (nickwhm) [reply src/components/templates/ClaimsList/index.tsx:126]: I don't think I can use index as a key you get the react error - " error  Do not use Array index in keys  react/no-array-index-key"

I've changed it to be first name as the key which should be ok as I doubt you'd claim for children with the same first name.

File: src/components/templates/ClaimsList/index.tsx [hunk 4]
```diff
@@ -25,132 +25,186 @@ export default function ClaimsList(props){
...
-                <p key={child.firstName}>
-                  {child.firstName && child.lastName && `${child.firstName} ${child.lastName}`}
-                  {child.dob && <><br/><span className='govuk-!-font-weight-regular'>{`${t('DATE_OF_BIRTH')} ${child.dob}`}</span><br/></>}
-                  <span className='govuk-!-font-weight-regular'>{t('CREATED_DATE')} {claimItem.dateCreated}</span>
-                </p>
-              )}
-              {claimItem.actionButton}
+  function renderChildDetails(claimItem) {
+    return claimItem.children.map((child, index) => (
+      <dl className='govuk-summary-list' key={child.firstName}>
```
Comment (gautamkumar-hmrc) [inline src/components/templates/ClaimsList/index.tsx:126]: I was suggesting you to append index with child name like key={child.firstName+index}>

Comment (nickwhm) [reply src/components/templates/ClaimsList/index.tsx:126]: I understand that. But if you do use the index at all as a key you get that error above when you run lint.

**Review by gautamkumar-hmrc — APPROVED** [review]

**Review by ModyGhaly — APPROVED** [review]

**Pull request merged**